lib/logstash/filters/jdbc/read_write_database.rb

```diff
@@ -0,0 +1,61 @@
+require_relative "basic_database"
+
+module LogStash module Filters module Jdbc
+  class ReadWriteDatabase < BasicDatabase
+
+    def repopulate_all(loaders)
+      case loaders.size
+        when 1
+          fill_and_switch(loaders.first)
+        when 2
+          fill_and_switch(loaders.first)
+          fill_and_switch(loaders.last)
+        else
+          loaders.each do |loader|
+            fill_and_switch(loader)
+          end
+      end
+    end
+
+    alias populate_all repopulate_all
+
+    def fetch(statement, parameters)
+      @rwlock.readLock().lock()
+      @db[statement, parameters].all
+    ensure
+      @rwlock.readLock().unlock()
+    end
+
+    def build_db_object(db_object)
+      @rwlock.writeLock().lock()
+      db_object.build(@db)
+    ensure
+      @rwlock.writeLock().unlock()
+    end
+
+    private
+
+    def fill_and_switch(loader)
+      records = loader.fetch
+      return if records.size.zero?
```

Comment:
We need to write something to the log, If we don't get any data from the server we won't be able to do any match? This would help debug bad queries OR bad behaviors of other systems.